src/mon/AuthMonitor.cc

```diff
@@ -1888,15 +1891,32 @@ AuthMonitor::caps_update AuthMonitor::_gen_wanted_caps(EntityAuth& e_auth,
     }
 
     if (cap_entity == "mon") {
-      is_caps_update_reqd = _merge_caps<MonCap>(cap_entity, new_cap_str,
+      is_caps_update_reqd_mon = _merge_caps<MonCap>(cap_entity, new_cap_str,
 	cur_cap_str, newcaps, out);
     } else if (cap_entity == "osd") {
-      is_caps_update_reqd = _merge_caps<OSDCap>(cap_entity, new_cap_str,
+      is_caps_update_reqd_osd = _merge_caps<OSDCap>(cap_entity, new_cap_str,
 	cur_cap_str, newcaps, out);
     } else if (cap_entity == "mds") {
-      is_caps_update_reqd = _merge_caps<MDSAuthCaps>(cap_entity, new_cap_str,
-	cur_cap_str, newcaps, out);
-    }
+      is_caps_update_reqd_mds = _merge_caps<MDSAuthCaps>(cap_entity,
+	new_cap_str, cur_cap_str, newcaps, out);
+    }
+  }
+
+  // if any one of MON, OSD or MDS caps failed to parse, it is pointless
+  // to run the update procedure.
+  if (is_caps_update_reqd_mon == CAPS_PARSING_ERR ||
+      is_caps_update_reqd_osd == CAPS_PARSING_ERR ||
+      is_caps_update_reqd_mds == CAPS_PARSING_ERR) {
+    is_caps_update_reqd = CAPS_PARSING_ERR;
+  // even if any one of MON, OSD or MDS caps needs an update, the update
+  // procedure needs to be executed.
+  } else if (is_caps_update_reqd_mon == CAPS_UPDATE_REQD ||
+      is_caps_update_reqd_osd == CAPS_UPDATE_REQD ||
+      is_caps_update_reqd_mds == CAPS_UPDATE_REQD) {
+    is_caps_update_reqd = CAPS_UPDATE_REQD;
+  // only if all caps don't require update, don't run the update procedure.
+  } else {
+    is_caps_update_reqd = CAPS_UPDATE_NOT_REQD;
```

Comment:
Is the `else` part needed - the default value for `is_caps_update_reqd` is `CAPS_UPDATE_NOT_REQD`.

Reply:
True, but this is just to be safe. In future, if there is a change between first line and last line due to which value of `is_caps_update_reqd` is disturbed, this line would save us from a bug.

Reply:
That code would be vetted during review, so , I don't think its required.

Reply:
Why isn't this taken care of @rishabh-d-dave esp when you requested a review of the changes?

Reply:
It's easy to miss such cases during review, so I thought it is good to have. But, anyways, I've removed it now.

Reply:
> It's easy to miss such cases during review, so I thought it is good to have. But, anyways, I've removed it now.

Sure. But I'd expect to be explicitly mentioned in the comment reply that its not been taken care of, else to me it looks like the change request has been ignored.